File: mangalam-wifi-zone/server/models/Plan.js
const mongoose = require("mongoose");

const planSchema = new mongoose.Schema(
  {
    key: {
      type: String,
      required: true,
      unique: true,
    },
    name: {
      type: String,
      required: true,
    },
    price: {
      type: Number,
      required: true,
    },
    durationMinutes: {
      type: Number,
      required: true,
    },
    visible: {
      type: Boolean,
      default: true,
    },
  },
  { timestamps: true }
);

module.exports = mongoose.model("Plan", planSchema);

File: flutter_app/lib/models/plan.dart
class Plan {
  final String id;
  final String name;
  final double price;
  final int durationMinutes;

  Plan({
    required this.id,
    required this.name,
    required this.price,
    required this.durationMinutes,
  });

  factory Plan.fromJson(Map<String, dynamic> json) {
    return Plan(
      id: json['_id'],
      name: json['name'],
      price: json['price'].toDouble(),
      durationMinutes: json['durationMinutes'],
    );
  }
}
